splunklib/searchcommands/internals.py

```diff
@@ -748,37 +772,36 @@ class RecordWriterV2(RecordWriter):
     def flush(self, finished=None, partial=None):
 
         RecordWriter.flush(self, finished, partial)  # validates arguments and the state of this instance
-        inspector = self._inspector
-
-        if self._flushed is False:
-
-            self._total_record_count += self._record_count
-            self._chunk_count += 1
-
-            # TODO: DVPL-6448: splunklib.searchcommands | Add support for partial: true when it is implemented in
-            # ChunkedExternProcessor (See SPL-103525)
-            #
-            # We will need to replace the following block of code with this block:
-            #
-            # metadata = [
-            #     ('inspector', self._inspector if len(self._inspector) else None),
-            #     ('finished', finished),
-            #     ('partial', partial)]
-
-            if len(inspector) == 0:
-                inspector = None
-
-            if partial is True:
-                finished = False
 
-            metadata = [item for item in (('inspector', inspector), ('finished', finished))]
-            self._write_chunk(metadata, self._buffer.getvalue())
-            self._clear()
+        if partial or not finished:
+            # Don't flush partial chunks, since the SCP v2 protocol does not
+            # provide a way to send partial chunks yet.
+            return
 
-        elif finished is True:
-            self._write_chunk((('finished', True),), '')
+        if not self.is_flushed:
+            self.write_chunk(finished=True)
 
-        self._finished = finished is True
+    def write_chunk(self, finished=None):
+        inspector = self._inspector
+        self._committed_record_count += self.pending_record_count
+        self._chunk_count += 1
+
+        # TODO: DVPL-6448: splunklib.searchcommands | Add support for partial: true when it is implemented in
+        # ChunkedExternProcessor (See SPL-103525)
+        #
```

Comment:
do we need to remove the bug#?

Reply:
Do you mean you think the bug has addressed and we can remove this comment? We are OK to refer to ticket numbers as long as we don't provide links if that's the concern.